test/test_proxymanager.py

```diff
@@ -57,3 +63,15 @@ def test_proxy_tunnel(self):
         with ProxyManager("https://proxy:8080") as p:
             assert p._proxy_requires_url_absolute_form(http_url)
             assert p._proxy_requires_url_absolute_form(https_url)
+
+    def test_proxy_connect_retry(self):
+        retry = Retry(total=None, connect=False)
+        with find_unused_port() as port:
+            with ProxyManager("http://localhost:{}".format(port)) as p:
+                with pytest.raises(ProxyError):
+                    p.urlopen("HEAD", url="http://localhost/", retries=retry)
+
+        retry = Retry(total=None, connect=2)
```

Comment:
We don't have an `assert` that the resulting `MaxRetryError` was caused by connection errors. Might be good to inspect the exception.

Reply:
I've added the lines to the test, is that what you meant?

Reply:
Yeah that looks good to me :)